src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,77 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int ARRAY_GROW_COEFFICIENT = 2;
+    private Node<K, V>[] table = new Node[DEFAULT_INITIAL_CAPACITY];
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = getHush(key) % table.length;
+        Node<K, V> newNode = new Node<>(key, value);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.key == null && key == null || key != null && key.equals(node.key)) {
```

Comment:
```suggestion
            if (node.key == key || key != null && key.equals(node.key)) {
```
create checkKey() method